app/views/declaration/border_transport.scala.html

```diff
@@ -26,77 +26,107 @@
+        tariffDetails: tariffDetails,
+        formHelper: uk.gov.hmrc.play.views.html.helpers.FormWithCSRF
+)
 
 @(mode: Mode, form: Form[BorderTransport])(implicit request: JourneyRequest[_], messages: Messages)
 
 @tariffBodyDesc = { @Html(messages("declaration.type.consignmentTariffText", components.details_content_link())) }
 
 @tariffBody = { @components.tariff_body_partial(messages("declaration.transportInformation.active.help-item1"), messages("declaration.transportInformation.active.help-item2"))(tariffBodyDesc)}
 
+@inputField(transportType: String, labelKey: String) = @{Some(exportsInputText(
+    field = form(s"borderTransportReference_$transportType"),
+    labelClasses = "govuk-label",
+    inputClasses = Some("govuk-input govuk-!-width-one-third"),
+    labelKey = labelKey
+))
+}
 
-@radioOption(id: String, value: String) = @{RadioOption(
-    id = id,
-    value = value,
-    message = messages(s"declaration.transportInformation.meansOfTransport.$id"),
-    revealComponent = Some(components.fields.field_text(
-        field = form(s"borderTransportReference_$id"),
-        label = Some(messages(s"declaration.transportInformation.meansOfTransport.$id.label"))
-    )
-))}
-
-@main_template(
-    title = Title("declaration.transportInformation.active.title")
-) {
-
-    @components.back_link(Navigator.backLink(BorderTransport, mode))
-
-    @helper.form(BorderTransportController.submitForm(mode), 'autoComplete -> "off") {
-        @helper.CSRF.formField
-
-        @components.error_summary(form.errors)
-
-        @components.heading(title = messages("declaration.transportInformation.active.title"))
-
-        @components.fields.field_radio(
-            field = form("borderTransportType"),
-            legend = messages("declaration.transportInformation.meansOfTransport.crossingTheBorder.header"),
-            hint = Some(messages("declaration.transportInformation.meansOfTransport.crossingTheBorder.header.hint")),
-            inputs = Seq(
-                radioOption("IMOShipIDNumber",IMOShipIDNumber),
-                radioOption("nameOfVessel",NameOfVessel),
-                radioOption("wagonNumber", WagonNumber),
-                radioOption("vehicleRegistrationNumber", VehicleRegistrationNumber),
-                radioOption("IATAFlightNumber", IATAFlightNumber),
-                radioOption("aircraftRegistrationNumber", AircraftRegistrationNumber),
-                radioOption("europeanVesselIDNumber", EuropeanVesselIDNumber),
-                radioOption("nameOfInlandWaterwayVessel", NameOfInlandWaterwayVessel)
-            )
-        )
+@radioOption(id: String, value: String) = @{RadioItem(
+    id = Some(id),
+    value = Some(value),
+    content = Text(messages(s"declaration.transportInformation.meansOfTransport.$id")),
+    conditionalHtml = inputField(id, s"declaration.transportInformation.meansOfTransport.$id.label"),
+    checked = form("borderTransportType").value.contains(value))}
```

Comment:
Is it worth to extract this as a component or it's only connected with this page? We will be able to reuse this part in more places.

Reply:
As I see Github complicated my comment with old changes 😂

Reply:
Maybe.  But as Paulo has approved it's going in as it is.